code/unclassified/singleNumber/single_number.cpp

```diff
@@ -0,0 +1,28 @@
+// Author: Karan Chawla
+// Date: 8/8/17
+// Part of Cosmos by OpenGenus Foundation
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+/* Function to find the unique number in an array */
+
+int singleNumber(vector<int>& A) 
+{
+    int result = A[0];
+    
+    for(int i = 1 ; i < A.size() ; i++)
+    {
+        result = result^A[i];  
```

Comment:
For your input, it will give 4 as answer. Did you intend to keep one unique number in the array?

Reply:
Yes, that's the idea. To detect the one unique number in the given array.

Reply:
According to your input, the output should be 2 and 6 👍

Reply:
Yep, you're correct. Edited the test case to include 2 in the `nums` array. Now, the only unique number is 6, which is the output of the program.